pkg/reconciler/cluster.go

```diff
@@ -123,6 +123,15 @@ func syncCluster(
 			cr.Status.State = string(clusterReady)
 		}
 		return nil
+	} else if state == clusterMembersError {
+		if cr.Status.State != string(clusterWarning) {
+			shared.LogInfo(
+				cr,
+				"warning",
+			)
+			cr.Status.State = string(clusterWarning)
+		}
+		return nil
```

Comment:
Thinking more about this now that I see actual code... this may not be how we want/need to handle an overall cluster warning state.

From a purely functional perspective, having this as the overall state passed out of syncRoles means that we lose information about whether any new members have been created and whether any members are in a transitional state that still needs processing. That can have bad consequences for a) whether we call syncMembers and b) what happens inside syncMembers.

From a user viewpoint, it also means that once any member gets into error state, we lose the "big picture" cluster state that tells us if any updating is going on.

It feels like this pushes us toward NOT representing this in the overall cluster state property, but instead having a separate top-level info message (or eventually/ideally an event list).